adal/src/telemetry/java/com/microsoft/aad/adal/HttpEvent.java

```diff
@@ -92,100 +92,54 @@ void setRequestIdHeader(final String requestIdHeader) {
-        String errorCode = null;
-        String subErrorCode = null;
-        String tokenAge = null;
-        String speRing = null;
-
-        if (headerVersion.equals("1")) {
-            // The expected delimiter count of the v1 header
-            final int delimCount = 4;
-
-            // Verify the expected format "<version>, <error_code>, <sub_error_code>, <token_age>, <ring>"
-            Pattern headerFmt = Pattern.compile("^[1-9]+\\.?[0-9|\\.]*,[0-9|\\.]*,[0-9|\\.]*,[^,]*[0-9\\.]*,[^,]*$");
-            Matcher matcher = headerFmt.matcher(xMsCliTelem);
-            if (!matcher.matches()) {
-                Logger.w(TAG, "", "", ADALError.X_MS_CLITELEM_MALFORMED);
-                return;
-            }
-
-            headerSegments = xMsCliTelem.split(",", delimCount + 1);
-
-            final int indexErrorCode = 1;
-            final int indexSubErrorCode = 2;
-            final int indexTokenAge = 3;
-            final int indexSpeInfo = 4;
-
-            // get the error_code
-            errorCode = headerSegments[indexErrorCode];
-
-            // get the sub_error_code
-            subErrorCode = headerSegments[indexSubErrorCode];
-
-            // get the token_age
-            tokenAge = headerSegments[indexTokenAge];
-
-            // get the spe_ring
-            speRing = headerSegments[indexSpeInfo];
-        } else { // unrecognized version
-            Logger.w(TAG, "Unexpected header version: " + headerVersion, null, ADALError.X_MS_CLITELEM_VERSION_UNRECOGNIZED);
-        }
-        // Set the extracted values on the HttpEvent
-        if (!StringExtensions.isNullOrBlank(errorCode) && !errorCode.equals("0")) {
-            setServerErrorCode(errorCode);
+        if (!StringExtensions.isNullOrBlank(cliTelemInfo.getServerErrorCode())
+                && !cliTelemInfo.getServerErrorCode().equals("0")) {
+            setServerErrorCode(cliTelemInfo.getServerErrorCode());
         }
 
-        if (!StringExtensions.isNullOrBlank(subErrorCode) && !subErrorCode.equals("0")) {
-            setServerSubErrorCode(subErrorCode);
+        if (!StringExtensions.isNullOrBlank(cliTelemInfo.getServerSubErrorCode())
+                && !cliTelemInfo.getServerSubErrorCode().equals("0")) {
+            setServerSubErrorCode(cliTelemInfo.getServerSubErrorCode());
         }
 
-        if (!StringExtensions.isNullOrBlank(tokenAge)) {
-            setRefreshTokenAge(tokenAge);
+        if (!StringExtensions.isNullOrBlank(cliTelemInfo.getRefreshTokenAge())) {
+            setRefreshTokenAge(cliTelemInfo.getRefreshTokenAge());
         }
 
-        if (!StringExtensions.isNullOrBlank(speRing)) {
-            setSpeRing(speRing);
+        if (!StringExtensions.isNullOrBlank(cliTelemInfo.getSpeRing())) {
+            setSpeRing(cliTelemInfo.getSpeRing());
         }
     }
 
     void setServerErrorCode(final String errorCode) {
-        setProperty(EventStrings.SERVER_ERROR_CODE, errorCode.trim());
+        if (null != errorCode) {
```

Comment:
nit: There's double checking of errorCode, before calling this you are already checking "StringExtensions.isNullOrBlank" which I assume should already catch if it is null.

Reply:
That check is in a different class, though - this is intentional so that `.trim()` *cannot* be called on a `null` reference

Reply:
I'd just personally prefer doing all the checks in here, but that's not important :)